t-update-filter-behavior.html

```diff
@@ -26,61 +26,52 @@
+        inventories.forEach(function(inventory) {
+            Object.keys(filtersMapping).forEach(function(key) {
+                var mappedData = filtersMapping[key];
+                var context, totalFareComponent;
+
                 if (mappedData.hierarchy) {
-                    var context = this._traverseToHierarchy(mappedData.hierarchy, inventories[i])
+                    context = this._traverseToHierarchy(mappedData.hierarchy, inventory);
                 } else {
-                    var context = inventories[i];
+                    context = inventory;
                 }
+
                 if (mappedData.executeFunctionName) {
-                    var totalFareComponent = this._executeFunctionByName(mappedData.executeFunctionName, this, context, mappedData, buildFilters);
+                    totalFareComponent = this._executeFunctionByName(mappedData.executeFunctionName, this, context, mappedData, buildFilters);
                 }
             }
-        }
-
+        });
 
         //update Filters existing filters
-        for (var i = 0; i < filters.length; i++) {
-            for(var filterCategory in buildFilters){
-                if (filterCategory == filters[i].category) {
-                    if (filters[i].type == "Range") {
-                        filters[i].items[0].value = buildFilters[filterCategory].min.value
-                        filters[i].items[0].displayValue = buildFilters[filterCategory].min.displayValue
-                        filters[i].items[1].value = buildFilters[filterCategory].max.value
-                        filters[i].items[1].displayValue = buildFilters[filterCategory].max.displayValue
-                    } else if(filters[i].type == "Options"){
-                        //Make count of options filter zero
-                        for (var j = 0; j < filters[i].items.length; j++) {
-                            if (filters[i].items[j].additionalInfo && filters[i].items[j].additionalInfo.length) {
-                                filters[i].items[j].additionalInfo[0].value = "0";
-                            }
-                        }
-                        
-                        //Add exact count
-                        for (var j = 0; j < filters[i].items.length; j++) {
-                            for(var filterLebel in buildFilters[filterCategory]){
-                                if (filters[i].items[j].label.toLowerCase() == filterLebel.toLowerCase()) {
-                                    if (filters[i].items[j].additionalInfo && filters[i].items[j].additionalInfo.length) {
-                                        filters[i].items[j].additionalInfo[0].value = buildFilters[filterCategory][filterLebel];
-                                    }
-                                }
-                            }
-                        }
+        filters.forEach(function(filter) {
+            var buildFilter = buildFilters[filter.category];
```

Comment:
We can get rid of the nested for loop iterating over object properties and directly see if there is a value for the key instead.